File: back_end/java_version/anti_patterns/src/main/java/com/shipay/challenge/middlewares/Exceptions.java
package com.shipay.challenge.middlewares;

public class Exceptions {
    public static class BadRequestException extends RuntimeException {
        public BadRequestException(String message) { super(message); }
    }

    public static class DataNotFoundException extends RuntimeException {
        public DataNotFoundException(String message) { super(message); }
    }

    public static class ExternalServiceException extends RuntimeException {
        public ExternalServiceException(String message) { super(message); }
    }
}

File: back_end/teahupoo/anti_patterns/Middlewares/Exceptions.cs
using System;

namespace Shipay.Challenge.Middlewares
{
    public class BadRequestException : Exception { public BadRequestException(string msg) : base(msg) {} }
    public class DataNotFoundException : Exception { public DataNotFoundException(string msg) : base(msg) {} }
    public class ExternalServiceException : Exception { public ExternalServiceException(string msg) : base(msg) {} }
}
